src/minter-suite-mapping.ts

```diff
@@ -692,54 +680,30 @@ export function handleUnregisteredNFTAddressV2(
 // Because AssemblyScript does not support union types, we need to manually type check inside each method, to ensure correct usage.
 // For any questions reach out to @jon or @ryley-o.eth. or see the following document https://docs.google.com/document/d/1XSxl04eJyTxc_rbj6cmq-j00zaYDzApBBLT67JXtaOw/edit?disco=AAAAZa8xp-Q
 
-export function handleSetValueGeneric<T, C>(
-  event: T,
-  config: C,
-  project: Project | null
+export function handleSetMinterDetailsGeneric<T, C>(
+  key: string,
+  value: T,
+  config: C
 ): void {
-  if (
-    !(
-      event instanceof ConfigValueSetBool ||
-      event instanceof ConfigValueSetBigInt ||
-      event instanceof ConfigValueSetAddress ||
-      event instanceof ConfigValueSetBytes
-    )
-  ) {
-    return;
-  }
-
-  if (
-    !(config instanceof ProjectMinterConfiguration || config instanceof Minter)
-  ) {
-    return;
-  }
-
-  if (project) {
-    project.updatedAt = event.block.timestamp;
-  }
-
   let minterDetails = getMinterDetails(config);
-  let jsonKey = event.params._key.toString();
+  let jsonKey = key;
   let jsonValue: JSONValue;
 
```

Comment:
should we add something like the following? (we might get errors when running the subgraph locally without?)
```
if (
    !(config instanceOf Minter ||
    config instanceOf Project)
) {
    log.warning("[WARN] Generic property attempted to be set on something not a Minter or Project")
    return;
}
```
Also, we might want to do something similar for value `<T>` (for the same reasons?)

Reply:
++ Since we're updating the entity here, I do think the type check logic should also be here. @ryley-o `T` is type narrowed below so we don't need to do the extra type check but should definitely still warn like suggested.

Overall, adding Ryley's logic is enough here but thinking on this more, we should do some additional refactoring as a follow-up that should simplify/clarify the code a bit! Currently, this function is sort of halfway between a utility function and an actual handler, which makes it a little hard to understand where and what to update. Instead, we can break out setJSONValue that takes `key`, `value`, and `map` and returns the updated map.  I think we'd want to do something similar for all of the handlers that are mostly doing `TypedMap<string, JSONValue>` manipulation so that we can use extra_minter_details for custom events without having to manipulate them to fit the generic event format.

Reply:
Sure! I [moved this type-checking](https://github.com/ArtBlocks/artblocks-subgraph/pull/135/commits/a697ac6670cda6d5e3a82f864de8064d44157acf#diff-a1077c7a5a21bf6e3a06b3124ce379bd5374fef2987d7164b250ae14fad15c56R729) outside this function (to our handler) because without it, the `project.updatedAt` call would run even if `handleSetMinterDetailsGeneric` bombed due to a failed type-check. 

Makes sense that we'd want this check to live inside the Generic handler so I added Ryley's logic + I'll add a ticket to refactor this too :)

Reply:
Asana ticket for follow-up refactor
https://app.asana.com/0/1202106026831420/1203450133533228